### DISCUSSION TITLE
Database functions with custom types in GraphQL

### DISCUSSION THREAD

**Discussion by joshjo** [supabase/supabase#21510] [Questions] [answered]
I am trying to create a database function with a custom type, but this endpoint does not appear in the GraphQL Schema. 

This is only happening with custom types. The function works if the function returns a SQL type or a SET of a created table.

```sql
CREATE TYPE nice_type AS (
  number int
);

DROP FUNCTION test_function;

CREATE OR REPLACE FUNCTION test_function() RETURNS SETOF nice_type AS
$$
  SELECT id FROM properties;
$$
LANGUAGE sql STRICT STABLE;
```

The function can be called using the SQL Editor. Also, it is available using the RPC client but not in the GraphQL Schema. 

<img width="1323" alt="Screenshot 2024-02-24 at 09 54 08" src="https://github.com/supabase/supabase/assets/1036364/f57ce6b9-7e11-4458-aa49-84a13417f0b5">

**Comment by GaryAustin1** [comment, marked as answer]
https://supabase.github.io/pg_graphql/functions/
Seems like not supported.  Only some return types are.
